src/sentry/api/endpoints/project_rules.py

```diff
@@ -49,7 +54,28 @@ def post(self, request, project):
         serializer = RuleSerializer(context={"project": project}, data=request.data)
 
         if serializer.is_valid():
-            rule = serializer.save(rule=Rule())
+            data = serializer.validated_data
+            kwargs = {
+                "name": data["name"],
+                "environment": data.get("environment", None),
+                "project": project,
+                "action_match": data["actionMatch"],
```

Comment:
Mapping these dicts manually across multiple places feels a bit unsafe, don't we have a helper to do this automatically? I remember @EvanPurkhiser creating a camelCase2snake_case serializer recently.

Reply:
This is the standard for now. If we want to make using `camelCase2snake_case` the standard, please propose it in a backend TSC meeting.

Reply:
I'm not proposing anything, just pointing out that doing this repeatedly is error prone.